src/server/transaction.cc

```diff
@@ -380,7 +382,11 @@ void Transaction::MultiSwitchCmd(const CommandId* cid) {
   cb_ = nullptr;
 
   if (multi_->mode == NON_ATOMIC) {
-    shard_data_.resize(0);
+    for (auto& sd : shard_data_) {
+      sd.arg_count = sd.arg_start = sd.local_mask = 0;
+      sd.pq_pos = TxQueue::kEnd;
+      sd.is_armed.store(false, memory_order_relaxed);
+    }
```

Comment:
I'd really like to default-assign it, but atomic is not copyable

Reply:
pls add DCHECK before store that is_armed.load(relaxed) is false?
(the store should be redundant anyway, imho).